### PR TITLE
Added query string parsing and a member to request

### PR DESCRIPTION
**Pull request by harella1**

### CODE REVIEW

**Comment by harella1** [comment]
An example to see query string as json:

```
#include <boost/property_tree/ptree.hpp>
#include <boost/property_tree/json_parser.hpp>
	http_server.resource["^/test"]["GET"] = [](shared_ptr<HttpServer::Response> response,
 shared_ptr<HttpServer::Request> request) {
		boost::property_tree::ptree qs;
		for (auto& qs_items : request->query_string) {
			qs.put<string>(qs_items.first, qs_items.second);
		}
		stringstream content;
		boost::property_tree::write_json(content, qs);
		content.seekp(0, ios::end);
		*response << "HTTP/1.1 200 OK\r\nContent-Length: " << content.tellp() << "\r\n\r\n" << content.rdbuf();
	};
```

**Comment by eidheim** [comment]
Thank you! I'll have a deeper look into this in two days or so.

**Comment by eidheim** [comment]
This PR took a bit more time than I expected. The reasons being:
* instead of populating the query_string automatically (and thus add a slight overhead on every request), one should consider creating a method (for instance called `parse_query` that instead returns the unordered_multimap. 
* The special characters (`%..` and `+`) are not converted to UTF-8 in the unordered_multimap. Hopefully this can be done without a large lookup table, but we would have to look into this. My hope is that the `%..` characters represent UTF-8 characters already.

File: server_http.hpp [hunk 1]
```diff
@@ -320,6 +321,24 @@ namespace SimpleWeb {
                     request->method=line.substr(0, method_end);
                     request->path=line.substr(method_end+1, path_end-method_end-1);
 
+					//search and populte query_string
+					size_t qs_start;
+					if ((qs_start = request->path.find('?')) != std::string::npos)
+					{
```
Comment (314159) [inline server_http.hpp]: I think this would be a bit cleaner:
`auto qs_start = request->path.find('?');`
`if (qs_start != std::string::npos)`
`{`

Comment (harella1) [inline server_http.hpp]: You may be right. I was just following the general style here, as in 
 `         if((method_end=line.find(' '))!=std::string::npos) {`
 `               if((path_end=line.find(' ', method_end+1))!=std::string::npos) {`

**Comment by harella1** [comment]
- Refactoring to 'parse_query' method makes sense, i'll try get to it soon.
- Special characters, maybe can be handled with the regex expression. 
- If a qs is present, the `request->path` should be changed to not include it, otherwise `find_resource` fails. I'll pr it too.

**Comment by eidheim** [comment]
Thank you for you latest commit. I probably have to bump the major version after merging this PR, since it has backwards compability issues (the path will change). I'll create a new release before this, and there are some minor changes also I want to do before version 3.0.

**Pull request merged**